recipes-core/swupdate/files/swupdate.install

```diff
@@ -0,0 +1,5 @@
+swupdate usr/bin
+swupdate.cfg etc
+swupdate_handlers.lua /usr/share/lua/5.3
```

Comment:
Why don't you bake the handler into the SWUpdate binary? Better than hard-coding the Lua path and more robust as well.

Reply:
On debian buster arm64 we have a build issue with embedded lua:
```
Output of:
aarch64-linux-gnu-gcc -Wall -Wshadow -Wwrite-strings -Wundef -Wstrict-prototypes -Wunused -Wunused-param
eter -Wunused-function -Wunused-value -Wmissing-prototypes -Wmissing-declarations -Wno-format-security -
Wno-format-truncation -Wold-style-definition -finline-limit=0 -fno-builtin-strlen -fomit-frame-pointer -
ffunction-sections -fdata-sections -fno-guess-branch-probability -funsigned-char -falign-functions=1 -fa
lign-jumps=1 -falign-labels=1 -falign-loops=1 -static-libgcc -fno-unwind-tables -fno-asynchronous-unwind
-tables -Os -I/usr/include/lua5.3 -Wl,-z,relro -Wl,-z,now -Wl,-E -Wl,--format=binary -Wl,swupdate_handle
rs.lua -Wl,--format=default -o tools/swupdate-hawkbitcfg -Wl,--sort-common -Wl,--sort-section,alignment 
-Wl,--gc-sections -Wl,--start-group tools/swupdate-hawkbitcfg.o corelib/lib.a ipc/lib.a mongoose/lib.a p
arser/lib.a suricatta/lib.a bootloader/lib.a -Wl,--end-group -Wl,--start-group -lpthread -llua5.3 -lconf
ig -larchive -lcrypto -lssl -lz -lubootenv -lsystemd -Wl,--end-group                                   
==========
collect2: fatal error: ld terminated with signal 11 [Segmentation fault]
compilation terminated.
make[1]: *** [Makefile:447: tools/swupdate-hawkbitcfg] Error 1
Failed: -Wl,--start-group -lpthread -llua5.3 -lconfig -larchive -lcrypto -lssl -lz -lubootenv -lsystemd 
-Wl,--end-group                                                                                        
Output of:
aarch64-linux-gnu-gcc -Wall -Wshadow -Wwrite-strings -Wundef -Wstrict-prototypes -Wunused -Wunused-param
eter -Wunused-function -Wunused-value -Wmissing-prototypes -Wmissing-declarations -Wno-format-security -
Wno-format-truncation -Wold-style-definition -finline-limit=0 -fno-builtin-strlen -fomit-frame-pointer -
ffunction-sections -fdata-sections -fno-guess-branch-probability -funsigned-char -falign-functions=1 -fa
lign-jumps=1 -falign-labels=1 -falign-loops=1 -static-libgcc -fno-unwind-tables -fno-asynchronous-unwind
-tables -Os -I/usr/include/lua5.3 -Wl,-z,relro -Wl,-z,now -Wl,-E -Wl,--format=binary -Wl,swupdate_handle
rs.lua -Wl,--format=default -o tools/swupdate-client -Wl,--sort-common -Wl,--sort-section,alignment -Wl,
--gc-sections -Wl,--start-group tools/swupdate-client.o corelib/lib.a ipc/lib.a mongoose/lib.a parser/li
b.a suricatta/lib.a bootloader/lib.a -Wl,--end-group -Wl,--start-group -lpthread -llua5.3 -lconfig -larc
hive -lcrypto -lssl -lz -lubootenv -lsystemd -Wl,--end-group                                           
==========
collect2: fatal error: ld terminated with signal 11 [Segmentation fault]
compilation terminated.

```

Reply:
https://github.com/gylstorffq/meta-iot2050/tree/embedded-lua contains the changes necessary for embedded lua.

Reply:
OK, then if that works over there, this has a dependency on your branch? Or to which version would you like to consolidate to?

Reply:
As the other branch updates the distro to bullseye to provide gcc-10/gcc-9 for the build. So this not an option.